design.md

```diff
@@ -0,0 +1,121 @@
+
+At a high level the API will have the following methods on the localhost using port 9090:
+
+### Run/Stop Methods
+`RunProcess(RunProcessRequest)`
+`StopProcess(StopProcessRequest)`
+
+### Get Information Methods
+`GetProcessStatus(ProcessStatusRequest)`
+`GetProcessOutput(ProcessOutputRequest)`
+
+The `RunProcess` method will return the process identifier for the command issued to the client, which will be used for the `StopProcess`, `GetProcessStatus`, and `GetProcessOutput` methods.
+
+### Future Improvements
+* Run the server in a separate environment, such as kubernetes, which can allow for defining high availability through replicaSets and would move the API out from the domain of the operating system being managed
+* Utilize some kind of data store (relational or not) to track processes instead of relying on files
+* Include resource management of some kind to keep clients from over-taxing the operating system commands are running on
+* Scope processes to users that map back to the clients on the operating system
+
+## Server Authentication
+
+### mTLS
+The service will utilize mTLS to authenticate and secure requests. For this implementation we will manually create a single local CA certificate that will be used to generate a server side certificate and key pair. These certificates and keys will be stored locally for the server.
+
+The server will enforce TLS version 1.3 for maximum security. The server will prefer the TLS13-AES-256-GCM-SHA384 cipher suite to have the best possible encryption standard while maintaining efficient performance. 
+
+### Scoping Requests
+In order to scope requests to each client we will create a user for each client, and that will be authenticated against the user in the certificate sent from the client. The user on the host will match the first part of the email, before the `@`, which will be compared against the email we receive from the client. 
+
+The client will hard code this for now. Ideally this would be set via a configuration file. 
+
+#### Future Improvements
+* Use a certificate provider for the CA instead of a locally generating one
+* Utilize client certificate details to implement more fine grained access
+* Implement a dedicated secret management solution for private keys, such as Vault or AWS Secret Manager (or any other secure key store)
+* Automate certificate management going forward to remove human error and allow for easier revocation and rotation
+* Add monitoring for certificate expiration and authentication failures
+
+### Library
+The library will primarily utilize the builtin os package in Go to interact with the Linux operating system and accomplish the actual job management functions.
+
+The library will primarily utilize the os/exec builtin, with some use of the os/user and base level os builtin packages.
```

Comment:
Nit: `builtin` in Go refers to the builtin funcs (append, make, len, cap, etc). I would 
```suggestion
The library will primarily utilize the standard library's os/exec package, with some use of the os/user and os packages.
```